service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -28,69 +35,198 @@
+
+            collection.getExtensionsByUrl(FASTING_STATUS_URL).stream().findFirst()
+                .ifPresent(extension -> {
+                    Type value = extension.getValue();
+                    if (value instanceof CodeableConcept) {
+                        pertinentInformationBuilder.fastingStatus((CodeableConcept) value);
+                    } else if (value instanceof Duration) {
+                        pertinentInformationBuilder.fastingDuration((Duration) value);
+                    }
+                });
+
+            if (collection.hasQuantity()) {
+                pertinentInformationBuilder.quantity(collection.getQuantity());
+            }
+
+            if (collection.hasBodySite()) {
+                pertinentInformationBuilder.collectionSite(collection.getBodySite());
+            }
         }
 
-        return TemplateUtils.fillTemplate(
-            SPECIMEN_COMPOUND_STATEMENT_TEMPLATE,
-            specimenCompoundStatementTemplateParameters.build()
-        );
+        specimen.getNote().forEach(note -> pertinentInformationBuilder.note(note.getText()));
+
+        return getCollectionDateTime(specimen)
+            .map(pertinentInformationBuilder::build)
+            .orElseGet(pertinentInformationBuilder::build);
     }
 
-    private String prepareEffectiveTimeElement(Specimen specimen) {
-        if (specimen.getCollection().hasCollectedDateTimeType()) {
-            return String.format(
-                EFFECTIVE_TIME_TEMPLATE,
-                DateFormatUtil.toHl7Format(specimen.getCollection().getCollectedDateTimeType())
-            );
+    private static class PertinentInformationBuilder {
+
+        private static final String FASTING_STATUS = "Fasting Status:";
+        private static final String FASTING_DURATION = "Fasting Duration:";
+        private static final String QUANTITY = "Quantity:";
+        private static final String COLLECTION_SITE = "Collection Site:";
+        private static final String COMMENT_PREFIX = "comment type - LAB SPECIMEN COMMENT(E271)";
+
+        private String pertinentInformation;
+
+        PertinentInformationBuilder() {
+            pertinentInformation = StringUtils.EMPTY;
         }
 
-        if (specimen.hasReceivedTime()) {
-            return String.format(
-                EFFECTIVE_TIME_TEMPLATE,
-                DateFormatUtil.toHl7Format(specimen.getReceivedTimeElement())
-            );
+        public void fastingStatus(CodeableConcept fastingStatus) {
+            CodeableConceptMappingUtils.extractTextOrCoding(fastingStatus)
+                .map(fastingStatusValue -> newLine(withSpace(FASTING_STATUS, fastingStatusValue), pertinentInformation))
+                .ifPresent(text -> pertinentInformation = text);
+        }
+
+        public void fastingDuration(Duration fastingDuration) {
+            pertinentInformation = newLine(
+                withSpace(FASTING_DURATION, fastingDuration.getValue().toString(), fastingDuration.getUnit()),
```

Comment:
should we check `fastingDuration.hasValue()` first?

Reply:
I changed the way of parsing the value to string, here and for quantity as well, added tests for them.